### PR TITLE
Additional Python constructors for JobSpec and ResourceSpecV1

### LINKED ISSUE #75: Constructors for JobSpecification and ResourceSpecification classes
In the context of the Python prototype implementation we began to [discuss](https://github.com/ExaWorks/jpsi-python/pull/2/files#r546603797) constructors for the Specification classes - those do not yet define constructors.  This issues moves that discussion to the specification scope.

### PR DESCRIPTION
**Pull request by hategan**
And now something that actually uses all that new javascript stuff. Also fixes #75.

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -688,8 +688,30 @@ JobSpec()
 
 Constructs an empty `JobSpec`.
 
-#### Methods
+<div class="lang-bindings">
+
+__Python__:
+
+In Python, the following additional constructor is defined:
```
Comment (SteVwonder) [inline specification.md:695]: The argument-less constructor and this constructor are mutually exclusive in Python, right?  Do we want to also put the empty JobSpec constructor in this lang-bindings class?  Or maybe tweak the wording:

```suggestion
In Python, the following constructor is defined in lieu of the argument-less constructor:
```

File: specification.md [hunk 2]
```diff
@@ -1124,6 +1146,25 @@ ResourceSpecV1()
 
 Constructs an empty `ResourceSpecV1` object.
 
+<div class="lang-bindings">
+
+__Python__:
+
+In Python, the following additional constructor is defined:
```
Comment (SteVwonder) [inline specification.md:1153]: Same comment as above

**Review by SteVwonder — COMMENTED** [review]
Generally LGTM!  The language bindings JS is very slick!  One comment about the multiple constructors in python.

File: specification.md (hunk 1 above)
Comment (hategan) [reply specification.md:695]: Thanks for catching this. Yes, I think the wording should be updated. The all-keyword-arg constructor can be called as nullary, but the text should make it clear that that's not an "additional constructor".

**Review by SteVwonder — APPROVED** [review]
LGTM!

File: specification.md [hunk 3]
```diff
@@ -688,8 +688,30 @@ JobSpec()
...
+__Python__:
+
+In Python, the default constructor is replaced with the following constructor:
+
+<a name="jobspec-_**"></a>
+```python
+JobSpec(name: str = None, executable: str = None,
+        arguments: List[str] = None, directory: Path = None,
+        inheritEnvironment: bool = True,
+        environment: Dict[str, str] = None, stdinPath: Path = None,
```
Comment (jameshcorbett) [inline specification.md:702]: Why camelCase rather than underscores?

Comment (hategan) [reply specification.md:702]: because... silliness. Will update.

**Review by andre-merzky — APPROVED** [review]
LGTM

**Pull request merged**